text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,384 @@
+
+The actual removal of code paths that support the _old behavior_ will be deferred to _after_ the new major version has
+reached _General Availability_.
+
+### Monolithic Packaging
+
+The AWS Construct Library will be released as a single artifact, instead of a collection of more than 100 libraries. The
+`monocdk-experiment` package has the tools necessary to re-package the existing _hypermodular_ libraries into a single
+packaging. Re-using that same process will allow keeping the difference between `master` and `next` as constrained as
+possible (reducing the need for manual intervention in forward-porting features).
+
+Consequently, the `monocdk-experiment` package will be renamed so that it is published and advertised as
+`@aws-cdk/stdlib` on the `master` branch. Availability of this library as part of CDK v1 is going to provide customers
+with a stepping stone to begin migrating their applications to CDK v2.
+
+Additionally, all modules that are re-packaged as part of `@aws-cdk/stdlib` will be marked as `"private": true` in the
+`next` branch, making these available _only_ via `@aws-cdk/stdlib` in CDK v2.
+
+## Documentation Updates
+
+Before formally announcing the availability of CDK v2 for early adopters, a documentation source must be availble to
+facilitate use of the new repository. The minimum bar should be availability of an API reference documentation for the
+v2 codebase that is usable.
+
+At the same time, the work to update the [AWS CDK User Guide] should be scoped and planned out. The updates to the User
+Guide must also include a migration guide to help developers move from CDK v1 over to v2. Special attention will be
+spent in ensuring the v2 release notes contain a clear and accurate summary of breaking changes and how users should
+update their applications.
+
+[aws cdk user guide]: https://docs.aws.amazon.com/cdk/latest/guide/home.html
+
+## Prerelease Publishing
+
+Once sufficient documentation is in place, and the framework-level changes have been implemented and integration tested,
+we will be able to start promoting the upcoming new release and point early adopters to experimental builds of CDK v2.
+This will allow gathering early feedback on the updated experience.
+
+A prerelease announcement will be published, formally announcing our plan to release AWS CDK v2 in the near future. This
+prerelease announcement should at least mention the following elements:
+
+- A provisional list of breaking changes expected to be made to the _AWS Construct Library_ (`@aws-cdk/stdlib`) APIs
+  during the [Construct Library Maintenance Tasks](#construct-library-maintenance-tasks) phase
+- A finalized proposal for the CDK [v1 maintenance plan](#ongoing-maintenance-plan), and a call for developers to
+  provide feedback on that
+- A minimal migration guide for bravehearts who are inclined to try and migrate their existing (**non-production**)
+  applications
+
+## Construct Library Maintenance Tasks
```

Comment:
I think I am leaning towards agreeing more and more with @rix0rrr on this. I think every breaking change needs to be vetted as part of this RFC and if it is approved it will be done. I am taking back my proposal to allow owners to just make breaking changes after the fork.

Reply:
> I think every breaking change needs to be vetted as part of this RFC and if it is approved it will be done.

Given the aspirational timeline, I reckon this means no Construct Library breaking changes (beyond dropping `@deprecated` APIs) will be included in v2. I don't see how library owners will be able to afford the due diligence of an RFC before we need to have a delivery plan signed off.

Reply:
We can give people a couple of weeks to propose minor breaking changes in their modules for review. Worst case we won't have any.

Reply:
I am hopeful that most desired changes are already captured through deprecations so I am also fine with just picking those up.